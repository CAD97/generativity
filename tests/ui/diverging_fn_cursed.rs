use generativity::{Id, make_guard};
use never_say_never::Never;

fn assert_eq_lt<'id>(_: Id<'id>, _: Id<'id>) {}

fn diverging_fn_cursed() -> (Never, Never) {
    make_guard!(g_a);
    make_guard!(g_b);

    let a: Id = g_a.into();
    let b: Id = g_b.into();

    assert_eq_lt(a, b);

    loop {}
}

fn main() {}
