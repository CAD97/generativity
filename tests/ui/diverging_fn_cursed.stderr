error: unreachable expression
 --> tests/ui/diverging_fn_cursed.rs:8:5
  |
8 |     make_guard!(g_b);
  |     ^^^^^^^^^^^^^^^^
  |     |
  |     unreachable expression
  |     any code following this expression is unreachable
  |
note: this expression has type `(!, !)`, which is uninhabited
 --> tests/ui/diverging_fn_cursed.rs:8:5
  |
8 |     make_guard!(g_b);
  |     ^^^^^^^^^^^^^^^^
note: the lint level is defined here
 --> tests/ui/diverging_fn_cursed.rs:8:5
  |
8 |     make_guard!(g_b);
  |     ^^^^^^^^^^^^^^^^
  = note: this error originates in the macro `make_guard` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unreachable expression
 --> tests/ui/diverging_fn_cursed.rs:7:5
  |
7 |     make_guard!(g_a);
  |     ^^^^^^^^^^^^^^^^
  |     |
  |     unreachable expression
  |     any code following this expression is unreachable
  |
note: this expression has type `(!, !)`, which is uninhabited
 --> tests/ui/diverging_fn_cursed.rs:7:5
  |
7 |     make_guard!(g_a);
  |     ^^^^^^^^^^^^^^^^
note: the lint level is defined here
 --> tests/ui/diverging_fn_cursed.rs:7:5
  |
7 |     make_guard!(g_a);
  |     ^^^^^^^^^^^^^^^^
  = note: this error originates in the macro `make_guard` (in Nightly builds, run with -Z macro-backtrace for more info)
